syntax = "proto3";
option java_package = "com.aylaasia.referenceapp.grpc";
import "Base.proto";
/**
登录相关接口
 */
service AuthService{

  /**
  发送验证码
  @NoAuth
   */
  rpc sendVerificationCode(VerificationCodeReq) returns (Result);

  /**
  登录
  @NoAuth
   */
  rpc login(LoginReq) returns (Token);

  /**
  刷新token
   */
  rpc refreshToken(RefreshTokenReq) returns (Token);
}

/**
发送验证码入参
 */
message VerificationCodeReq{
  /**
  手机号
   */
  string phone = 1;
}

/**
登录入参
 */
message LoginReq{
  /**
  手机号
   */
  string phone = 1;
  /**
  验证码
   */
  string verificationCode = 2;
}

/**
登录出参
 */
message Token{
  /**
  token
   */
  string authToken = 1;
  /**
  刷新token
   */
  string refreshToken = 2;
  /**
  过期时间
   */
  int64 expireTime = 3;
}

/**
刷新token入参
 */
message RefreshTokenReq{
  /**
  刷新token
   */
  string refreshToken = 1;
}
